integration/localnet/bootstrap.go

```diff
@@ -78,39 +86,92 @@ func init() {
 }
 
+func generateBootstrapData(flowNetworkConf testnet.NetworkConfig) []testnet.ContainerConfig {
+	// Prepare localnet host folders, mapped to Docker container volumes upon `docker compose up`
+	prepareCommonHostFolders()
+	_, _, _, flowNodeContainerConfigs, _, err := testnet.BootstrapNetwork(flowNetworkConf, BootstrapDir)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Flow test network bootstrapping data generated...")
+	return flowNodeContainerConfigs
+}
+
+// localnet/bootstrap.go generates a docker compose file with images configured for a
+// self-contained Flow network, and other peripheral services, such as Observer services.
+// Private/Public keys and data are removed and re-created for the self-contained localnet
+// environment with every invocation, and are intended solely for testing, not for production.
 func main() {
 	flag.Parse()
 
-	fmt.Println("Bootstrapping a new FLITE network...")
+	// Prepare test node configurations of each type, access, execution, verification, etc
+	flowNodes := prepareFlowNodes()
 
-	fmt.Printf("Node counts:\n")
-	fmt.Printf("- Collection: %d\n", collectionCount)
-	fmt.Printf("- Consensus: %d\n", consensusCount)
-	fmt.Printf("- Execution: %d\n", executionCount)
-	fmt.Printf("- Verification: %d\n", verificationCount)
-	fmt.Printf("- Staked Access: %d\n", accessCount)
-	fmt.Printf("- Unstaked Access: %d\n\n", unstakedAccessCount)
-
-	nodes := prepareNodes()
-
-	opts := []testnet.NetworkConfigOpt{testnet.WithClusters(nClusters)}
+	// Generate a Flow network config for localnet
+	flowNetworkOpts := []testnet.NetworkConfigOpt{testnet.WithClusters(nClusters)}
 	if numViewsEpoch != 0 {
-		opts = append(opts, testnet.WithViewsInEpoch(numViewsEpoch))
+		flowNetworkOpts = append(flowNetworkOpts, testnet.WithViewsInEpoch(numViewsEpoch))
 	}
 	if numViewsInStakingPhase != 0 {
-		opts = append(opts, testnet.WithViewsInStakingAuction(numViewsInStakingPhase))
+		flowNetworkOpts = append(flowNetworkOpts, testnet.WithViewsInStakingAuction(numViewsInStakingPhase))
 	}
 	if numViewsInDKGPhase != 0 {
-		opts = append(opts, testnet.WithViewsInDKGPhase(numViewsInDKGPhase))
+		flowNetworkOpts = append(flowNetworkOpts, testnet.WithViewsInDKGPhase(numViewsInDKGPhase))
+	}
+	flowNetworkConf := testnet.NewNetworkConfig("localnet", flowNodes, flowNetworkOpts...)
+	displayFlowNetworkConf(flowNetworkConf)
+
+	// Generate the Flow network bootstrap files for this localnet
+	flowNodeContainerConfigs := generateBootstrapData(flowNetworkConf)
+
+	// Generate Flow network services docker-compose-nodes.yml
+	dockerServices := make(Services)
+	dockerServices = prepareFlowServices(dockerServices, flowNodeContainerConfigs)
+	serviceDisc := prepareServiceDiscovery(flowNodeContainerConfigs)
+	err := writePrometheusConfig(serviceDisc)
+	if err != nil {
+		panic(err)
 	}
-	conf := testnet.NewNetworkConfig("localnet", nodes, opts...)
 
+	// Generate the Observer services docker-compose-nodes.yml
```

Comment:
this comment probably isn't needed. it seems to imply observers get a different yml file

Reply:
Removed, thank you.